sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/protobuf/RazorMetricsImporterTest.java

```diff
@@ -74,25 +81,73 @@ public void roslyn_metrics_are_imported() {
-    verify(fileLinesContext).setIntValue(CoreMetrics.EXECUTABLE_LINES_DATA_KEY, 13, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.EXECUTABLE_LINES_DATA_KEY, 23, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.EXECUTABLE_LINES_DATA_KEY, 24, 1);
-
-    verify(fileLinesContext, times(2)).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 3, 1);
-    verify(fileLinesContext, times(2)).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 5, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 8, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 9, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 13, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 16, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 18, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 19, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 21, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 22, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 23, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 24, 1);
-    verify(fileLinesContext).setIntValue(CoreMetrics.NCLOC_DATA_KEY, 25, 1);
+    verifyMetrics(fileLinesContext, CoreMetrics.EXECUTABLE_LINES_DATA_KEY, 8, 23, 24);
+    verifyMetrics(fileLinesContext, CoreMetrics.NCLOC_DATA_KEY, 3, 5, 8, 9, 13, 16, 18, 19, 21, 22, 23, 24, 25);
+
+    assertThat(logTester.logs(Level.DEBUG)).isEmpty();
+  }
+
+  @Test
+  public void roslyn_metrics_are_imported_starting_with_4_10() throws FileNotFoundException {
+    var inputFile = addTestFileToContext("Cases.razor");
+    var noSonarFilter = mock(NoSonarFilter.class);
+    var fileLinesContext = mock(FileLinesContext.class);
+    var fileLinesContextFactory = mock(FileLinesContextFactory.class);
+    when(fileLinesContextFactory.createFor(any(InputFile.class))).thenReturn(fileLinesContext);
+
+    new MetricsImporter(sensorContext, fileLinesContextFactory, noSonarFilter, RazorImporterTestBase::fileName).accept(PROTOBUF_4_10_FILE.toPath());
+
+    var measures = sensorContext.measures(inputFile.key());
+    assertThat(measures).hasSize(7);
+
+    assertThat(measures).extracting("metric", "value")
+      .containsOnly(
+        Tuple.tuple(CoreMetrics.COMPLEXITY, 5),
+        Tuple.tuple(CoreMetrics.FUNCTIONS, 3),
+        Tuple.tuple(CoreMetrics.COMMENT_LINES, 0),
+        Tuple.tuple(CoreMetrics.COGNITIVE_COMPLEXITY, 1),
+        Tuple.tuple(CoreMetrics.CLASSES, 0),
+        Tuple.tuple(CoreMetrics.NCLOC, 14),
+        Tuple.tuple(CoreMetrics.STATEMENTS, 3));
+
+    verify(noSonarFilter).noSonarInFile(inputFile, Collections.emptySet());
+
+    verifyMetrics(fileLinesContext, CoreMetrics.EXECUTABLE_LINES_DATA_KEY, 8, 23, 24);
+    verifyMetrics(fileLinesContext, CoreMetrics.NCLOC_DATA_KEY, 1, 3, 5, 8, 9, 13, 16, 18, 19, 21, 22, 23, 24, 25);
```

Comment:
Why does this include `1` on the non-comment lines here?
Line#1 is the `@page` directive. 
Is this just roslyn internal change or should we care about this?

Reply:
Good catch! I think that's not correct and that it's caused by the generated code that has this snipped:
```
#nullable restore
#line (1,7)-(1,15) "\WebProject\Cases.razor"
"/cases"
```
I'll need to look into it and report an issue.

Reply:
I think in this PR we cannot handle that.

Reply:
Sure, no problem on my side. 
Resolving.

Reply:
See: https://github.com/dotnet/razor/issues/10375#issuecomment-2124657614